tests/data/test_validation.py

```diff
@@ -14,20 +9,38 @@ class TestValidation(unittest.TestCase):
     def test_coverage_json_valid(self):
         """p3.data.validation.coverage_json_valid"""
         json_string = '[{"file": "path", "id": "sha", "lines": [1, 2, [3, 5]]}]'
-        result_object = _validate_coverage_json(json_string)
+        result_object = _validate_coverage_data(json_string)
         expected_object = [{"file": "path", "id": "sha", "lines": [1, 2, [3, 5]]}]
-        self.assertTrue(result_object == expected_object)
+        self.assertEqual(result_object, expected_object)
 
     def test_coverage_json_invalid(self):
         """p3.data.validation.coverage_json_invalid"""
         json_string = '[{"file": "path", "id": "sha", "lines": [["1"]]}]'
         with self.assertRaises(ValueError):
-            _validate_coverage_json(json_string)
+            _validate_coverage_data(json_string)
 
-        with self.assertRaises(TypeError):
-            json_object = [{"file": "path", "id": "sha", "lines": [["1"]]}]
-            _validate_coverage_json(json_object)
+        invalid_json_string = '[{"file": "path", "id": "sha", "lines": [1, 2, [3, 5]]'
+        with self.assertRaises(ValueError):
+            _validate_coverage_data(invalid_json_string)
+
+    def test_coverage_object_valid(self):
+        """p3.data.validation.coverage_object_valid"""
+        json_object = [{"file": "path", "id": "sha", "lines": [1, 2, [3, 5]]}]
+        print(f"Testing valid object: {json_object}")
+        result_object = _validate_coverage_data(json_object)
```

Comment:
This test currently fails.  I think this is related to the comment I made above; you introduced a check that the coverage data is a `dict`, but here you can see that `json_object` is an array.  I suspect the test will start passing when the comment above is addressed, but I wanted to call it out here as well.